### PR TITLE
Feat: [s][c][t] Add xml resource plugin

### LINKED ISSUE #371: Add XML resources
**Is your feature request related to a problem? Please describe.**
I would like to automate XML file updates

**Describe the solution you'd like**
After looking to various libraries, [Dasel](https://github.com/TomWright/dasel) sounds very promising, even though additional work may be needed.

**Describe alternatives you've considered**
-

**Additional context**
* [Dasel](https://github.com/TomWright/dasel)

### PR DESCRIPTION
**Pull request by olblak**
Signed-off-by: Olblak <me@olblak.com>

Fix #371 

This pullrequest is a second attempt after https://github.com/updatecli/updatecli/pull/336

This time I am leveraging https://github.com/beevik/etree library

It went smoothly, and after some testing, I didn't notify any major change  done to the xml file after an update

Accept spec parameters are:

 * file: Define the xml file to manipulate
 * path: Specify the xml path query
 * value: Specify the xml value

## Test

To test this pull request, you can run the following commands:

```shell
cd pkg/plugins/resources/xml
go test
```

The path accept xpath query such as https://docs.python.org/3/library/xml.etree.elementtree.html#supported-xpath-syntax:w
Note that I didn't test all of them.

### Tradeoff

/

### Potential improvement
/

### CODE REVIEW

**Comment by olblak** [comment]
@jetersen I may have found an integrated solution

**Comment by olblak** [comment]
Combined with the autodiscovery feature introduced in "https://github.com/updatecli/updatecli/pull/763", we could automate pom.xml

**Comment by jetersen** [comment]
If you want we can easily test it Jenkinsci/bom since manifests are generated

**Comment by olblak** [comment]
Seems to working. I created this pullrequest https://github.com/olblak/bom/pull/1/files
using the following file

```
scms:
  default:
    kind: github
    spec:
      user: updatecli
      email: me@olblak.com
      owner: olblak
      repository: bom 
      token: '{{ requiredEnv "UPDATECLI_GITHUB_TOKEN" }}'
      username: '{{ requiredEnv "UPDATECLI_GITHUB_ACTOR" }}'
      branch: master
 
sources:
  jenkins-weekly:
    name: Get Latest Jenkins weekly version
    kind: jenkins
    spec:
      release: weekly

conditions:
  jenkins-weekly:
    name: Ensure bom properties is set to jenkins weekly
    kind: xml 
    scmid: default
    disablesourceinput: true
    spec:
      file: sample-plugin/pom.xml
      path: "./project/properties/bom"
      value: weekly
  
targets:
  jenkins-weekly:
    name: Bump Jenkins weekly version
    kind: xml 
    scmid: default
    spec:
      file: sample-plugin/pom.xml
      path: "./project/properties/jenkins.version"

pullrequests:
  default:
    title: '[updatecli] Bump Jenkins Weekly version to {{ source "jenkins-weekly" }}'
    kind: github
    scmid: default
    # Will soon be dropped, once https://github.com/updatecli/updatecli/pull/763 is merged
    targets:
      - jenkins-weekly

    spec:
      labels:
        - dependencies
        - skip-changelog
```

File: pkg/plugins/resources/xml/testdata/data_2.xml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<name>
+    <firstname tag="test" >John</firstname>
+    <lastname>Doe</lastname>
+</name>
```
Comment (olblak) [inline pkg/plugins/resources/xml/testdata/data_2.xml:5]:
```suggestion
</name>

```

**Comment by olblak** [comment]
@hervelemeur If you have some time, I would appreciate your review on this PR as you are pretty good at catching things I don't see

**Comment by olblak** [comment]
Imho the PR is ready to be merged

**Comment by olblak** [comment]
I am moving forward so I can already start using this in production

**Comment by lemeurherve** [comment]
> @hervelemeur If you have some time, I would appreciate your review on this PR as you are pretty good at catching things I don't see

Erf, I had some suggestions indeed, too late.

**Pull request merged**